src/cascadia/TerminalSettingsEditor/Appearances.cpp

```diff
@@ -20,6 +20,28 @@ using namespace winrt::Microsoft::Terminal::Settings::Model;
 
 namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
 {
+    bool Font::HasPowerlineCharacters()
+    {
+        if (!_hasPowerlineCharacters.has_value())
+        {
+            try
+            {
+                winrt::com_ptr<IDWriteFont> font;
+                THROW_IF_FAILED(_family->GetFont(0, font.put()));
+                BOOL exists{};
+                // We're actually checking for the "Extended" PowerLine glyph set.
+                // They're more fun.
+                THROW_IF_FAILED(font->HasCharacter(0xE0B6, &exists));
```

Comment:
It's worth noting that you aren't really checking for PowerLine glyphs here. This is a Private Use Area, and this particular range has been used for Cirth characters in the ConScript Unicode Registry for the past couple of decades. Admittedly there probably aren't a lot of terminal users with Cirth fonts, but it would be nice if there was a safer way to detect PowerLine fonts specifically. Could you not guesstimate that from the name or something?

Reply:
> Could you not guesstimate that from the name or something?

That was my first approach! Man, the code was a lot shorter then.

I profiled a bunch of fonts and found that there's:

- fonts with "PL" in them
	- at the end
	- in the middle ("Xxx PL Mono")
    	- ... which don't have PowerLine glyphs either ("IBM Plex Mono")
- NerdFonts
    - with "NF" at the end or in the middle ("Hack NF")
    - with "Nerd Font" at the end or in the middle ("Xxx Nerd Font Propo")
    - with "NerdFont" in them somewhere
- A hypothetical category of fonts that don't include a designator in the name, but do have glyphs there

I know that it'll always be "best effort", and that powerline has coöpted part of the PUA that has already been used, but... it sucks all-up no matter how you slice it.

Reply:
> Admittedly there probably aren't a lot of terminal users with Cirth fonts

![image](https://github.com/microsoft/terminal/assets/18356694/8e9bb299-bfa3-4f2a-9e6c-92840cc1f7e5)

Reply:
My thinking was that it wouldn't matter that much if you failed to detect some nerd fonts, because they'd just get the regular preview, which isn't the end of the world. Showing the nerd font preview to someone that's using another glyph set seems like a bigger deal.

That said, I've just been looking for a Cirth font to test with, and I couldn't find _any_ monospace variants that could reasonably be used in a terminal (I'm sure I had one in the past, but I might be misremembering that). So feel free to leave it as is. I just wanted to make sure we at least considered other options here.

Reply:
I personally think the codepoint check is more robust, because that's how most terminals detect powerline glyphs anyways. VS Code for instance checks these codepoints to determine if it should increase the contrast of text or not (it won't change the colors for powerline glyphs). Given that it's somewhat unlikely for a user to use a Cirth font and much more likely for them to use a powerline font, I believe that the current approach is better and more robust.